redis-lettuce/src/main/java/io/micronaut/configuration/lettuce/AbstractRedisClientFactory.java

```diff
@@ -48,10 +49,13 @@ public RedisClient redisClient(AbstractRedisConfiguration config) {
      *
      * @param config The configuration
      * @param optionalClientResources The ClientResources
+     * @param mutators The list of mutators
      * @return The {@link RedisClient}
      */
-    public RedisClient redisClient(AbstractRedisConfiguration config, @Nullable ClientResources optionalClientResources) {
-        ClientResources clientResources = configureClientResources(config, optionalClientResources);
+    public RedisClient redisClient(AbstractRedisConfiguration config,
+                                   @Nullable ClientResources optionalClientResources,
+                                   @Nullable List<ClientResourcesMutator> mutators) {
+        ClientResources clientResources = configureClientResources(config, optionalClientResources, mutators);
```

Comment:
You can't change this constructor because that's a breaking change. You need to add a new one, annotate it with `@Creator` and deprecate this new one.

Reply:
@ilopmar do you mean deprecating the old one (the one with 2 parameters) and annotating the new one (with 3 params) with `@Creator`?

Also, could you clarify why this is a breaking change? The new parameter is nullable

Reply:
Umm... two comments:
- You don't need to annotate it with `@Creator` because this method is not a constructor. My bad.
- It is a breaking change because you are modifying a public method in the factory. If any user is extending from this factory the change will break their code.

Reply:
Fair enough, will do. Thanks for the clarification!

Reply:
done!